QuizApp/Screens/HomePage/GPAView.swift

```diff
@@ -0,0 +1,171 @@
+        stackView.alignment = .center
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        return stackView
+    }()
+    
+    private let gpaLabel: UILabel = {
+        let label = UILabel()
+        label.text = Constants.HomePageConstants.gpaText
+        label.textColor = Constants.Colors.neutralWhite
+        label.font = .systemFont(ofSize: Constants.FontSizes.medium)
+        return label
+    }()
+    
+    private let gpaDigit: UILabel = {
+        let label = UILabel()
+        label.text = Constants.HomePageConstants.gpaDigit
+        label.textColor = Constants.Colors.yellowPrimary
+        label.font = .systemFont(ofSize: Constants.FontSizes.medium)
+        return label
+    }()
+    
+    private let detailsButton: UIButton = {
+        let button = UIButton()
+        button.setTitle(Constants.HomePageConstants.detailsButtonText, for: .normal)
+        button.setTitleColor(Constants.Colors.neutralWhite, for: .normal)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    private let nextArrow: UIImageView = {
+        let imageView = UIImageView()
+        imageView.contentMode = .scaleAspectFit
+        imageView.image = UIImage(named: Constants.Images.nextArrow)
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+    
+    // MARK: - Initializers
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - Setup Methods
+    
+    private func setupView() {
+        translatesAutoresizingMaskIntoConstraints = false
+        setupHierarchy()
+        setupConstraints()
+    }
+    
+    private func setupHierarchy() {
+        gpaStackView.addArrangedSubviews(
+            gpaLabel,
+            gpaDigit
+        )
+        
+        addSubview(gpaView)
+        gpaView.addSubviews(
+            gpaClearView,
+            gpaStackView,
+            detailsButton,
+            nextArrow
+        )
+    }
+    
+    private func setupConstraints() {
+        setupGPAViewConstraints()
+        setupGPAClearViewConstraints()
+        setupGPAStackViewConstraints()
+        setupDetailsButtonConstraints()
+        setupNextArrowConstraints()
+    }
+
+    // MARK: - Constraints Setup Methods
+    
+    private func setupGPAViewConstraints() {
+        NSLayoutConstraint.activate([
+            gpaView.topAnchor.constraint(equalTo: topAnchor),
+            gpaView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            gpaView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            gpaView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            gpaView.heightAnchor.constraint(equalToConstant: Constants.HomePageConstants.gpaViewHeight)
+        ])
```

Comment:
do you really need to apply six constraints. Check it and remove if any is unnecessary

Reply:
as i checked i need all constraints